### ISSUE TITLE
Should we have a global `CHPL_MEM` and `CHPL_JEMALLOC`?

### ISSUE THREAD

**Issue by jabraham17** [chapel-lang/chapel#25359]
Currently, we have the following environment variables for configuring Chapels memory allocation
- `CHPL_MEM`: the allocator to use for compiled chapel programs. Either `cstdlib` or `jemalloc`
  - `CHPL_TARGET_MEM`: same as `CHPL_MEM`, specifying both gives a warning and takes the value from `CHPL_TARGET_MEM`
- `CHPL_HOST_MEM`: the allocator to use for the chapel compiler. Either `cstdlib` or `jemalloc`
- `CHPL_JEMALLOC`: when `CHPL_MEM`/`CHPL_TARGET_MEM` is set to `jemalloc`, switches between `system` and `jemalloc`
  - `CHPL_TARGET_JEMALLOC`: same as `CHPL_JEMALLOC`, specifying both gives a warning and takes the value from `CHPL_TARGET_JEMALLOC`
- `CHPL_HOST_JEMALLOC`: when `CHPL_HOST_MEM` is set to `jemalloc`, switches between `system` and `jemalloc`

The reason for the duplicate names `CHPL_MEM`/`CHPL_TARGET_MEM` and `CHPL_JEMALLOC`/`CHPL_TARGET_JEMALLOC` is to provide backwards compatibility after https://github.com/chapel-lang/chapel/pull/18627. Note that `CHPL_JEMALLOC` is not a user facing feature and `CHPL_MEM` has been [documented to be deprecated in favor of `CHPL_TARGET_MEM`](https://chapel-lang.org/docs/2.1/usingchapel/chplenv.html#chpl-mem) since that PR.

Do we still want to support `CHPL_MEM`/`CHPL_JEMALLOC`?

The simple thing to do is to just remove `CHPL_MEM` and `CHPL_JEMALLOC`, however another possibility brought up on #25347 is to keep these variables, but just change their behavior slightly. `CHPL_MEM` would specify the default for `CHPL_TARGET_MEM` and `CHPL_HOST_MEM`, such that if `CHPL_MEM` is set to `jemalloc` then both `TARGET` and `HOST` are `jemalloc`. And similarly for `CHPL_JEMALLOC`. This would provide a shortcut for specifying the host and target memory configuration for users.

See https://github.com/chapel-lang/chapel/issues/25347#issuecomment-2187680628 for more details on how these defaults would work.

**Comment by jabraham17** [comment]
I agree that it would be a non-breaking change to do the simple thing for now and add the umbrella variables later if-needed. This would be my preference.

I do think that if we take that route, before we remove `CHPL_JEMALLOC` and `CHPL_MEM`, we should have warnings if these exist in an users environment at all, telling users to switch to the undeprecated terms. Just because our docs say `CHPL_MEM` is deprecated doesn't mean users have read that :).

**Comment by bradcray** [comment]
I agree, and was imagining printchplenv (or something in the build process) print the warning rather than just relying on docs in doing the deprecation.  Why don't we proceed with that plan if nobody speaks up organically on this issue in the meantime?
